ketcher-autotests/tests/Structure-Creating-&-Editing/SMARTS-attributes/Atom-properties-attributes/atom-properties-attributes.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { Page, test } from '@playwright/test';
+import {
+  BondTypeName,
+  clickInTheMiddleOfTheScreen,
+  doubleClickOnAtom,
+  pressButton,
+  selectBond,
+  takeEditorScreenshot,
+  waitForAtomPropsModal,
+  waitForPageInit,
+} from '@utils';
+import {
+  checkSmartsValue,
+  setAtomicMass,
+  setCharge,
+  setLabel,
+  setValence,
+} from '../utils';
+
+async function drawStructure(page: Page, numberOfClicks: number) {
+  await selectBond(BondTypeName.Single, page);
+  for (let i = 0; i < numberOfClicks; i++) {
+    await clickInTheMiddleOfTheScreen(page);
+  }
+}
+
+test.describe('Checking atom properties attributes in SMARTS format', () => {
+  const defaultFileFormat = 'MDL Molfile V2000';
+
+  test.beforeEach(async ({ page }) => {
+    const numberOfAtom = 0;
+    const numberOfBondsAtStructure = 3;
+    await waitForPageInit(page);
+    await drawStructure(page, numberOfBondsAtStructure);
+    await page.keyboard.press('Escape');
+    await doubleClickOnAtom(page, 'C', numberOfAtom);
+    await waitForAtomPropsModal(page);
+  });
+
+  test('Setting atom label and checking the atom number', async ({ page }) => {
+    await setLabel(page, 'Cr');
+    await pressButton(page, 'Apply');
+    await takeEditorScreenshot(page);
+    await checkSmartsValue(page, defaultFileFormat, '[#6](-[#6])(-[#24])-[#6]');
+  });
+
+  test('Setting positive charge', async ({ page }) => {
+    await setCharge(page, '10');
+    await pressButton(page, 'Apply');
+    await takeEditorScreenshot(page);
+    await checkSmartsValue(
+      page,
+      defaultFileFormat,
+      '[#6](-[#6])(-[#6;+10])-[#6]',
+    );
+  });
+
+  test('Setting negative charge', async ({ page }) => {
+    await setCharge(page, '-15');
+    await pressButton(page, 'Apply');
```

Comment:
The code is often repeated. Could you write a function and use it throughout the code where possible?)
For example:
async function setAndCheckAtomProperties(page, property, value, expectedSmarts) {
  await property(page, value);
  await pressButton(page, 'Apply');
  await takeEditorScreenshot(page);
  await checkSmartsValue(page, defaultFileFormat, expectedSmarts);
}

// Tests using the helper function
test('Setting atom label and checking the atom number', async ({ page }) => {
  await setAndCheckAtomProperties(page, setLabel, 'Cr', '[#6](-[#6])(-[#24])-[#6]');
});

test('Setting positive charge', async ({ page }) => {
  await setAndCheckAtomProperties(page, setCharge, '10', '[#6](-[#6])(-[#6;+10])-[#6]');
});

test('Setting negative charge', async ({ page }) => {
  await setAndCheckAtomProperties(page, setCharge, '-15', '[#6](-[#6])(-[#6;-15])-[#6]');
});

test('Setting atomic mass', async ({ page }) => {
  await setAndCheckAtomProperties(page, setAtomicMass, '30', '[#6](-[#6])(-[#6;30])-[#6]');
});

Reply:
Done